src/transformers/generation_utils.py

```diff
@@ -478,6 +478,11 @@ def _expand_inputs_for_generation(
         if attention_mask is not None:
             model_kwargs["attention_mask"] = attention_mask.index_select(0, expanded_return_idx)
 
+        lm_attention_mask = model_kwargs.get("lm_attention_mask", None)
+        if lm_attention_mask is not None:
+            if lm_attention_mask.shape[0] != 1:
```

Comment:
why do we need this? This is also relevant for `batch_size > 1` no?

Reply:
So `(lm_attention_mask.shape[0] != 1) == (batch_size > 1)` is should be almost always true. The only time it doesn't match is when we use the same mask for all elements on the match, and thus we allow broadcast for tensors (the same way we will allow broadcast along head dimension). In that specific case, if we can broadcast accross batch_size, we can probably do so along the expanded_batch_size as well.

I'm guessing what you mean is why do I not apply it when `batch_size == 1`? This is also to leverage broadcast as all elements in the expanded_batch should all apply the same mask.